resource-sharing-partners/src/main/java/no/sikt/rsp/ResourceSharingPartnerHandler.java

```diff
@@ -82,15 +85,35 @@ public Integer handleRequest(S3Event s3event, Context context) {
             logger.info("done collecting instreg");
             AlmaCodeProvider almaCodeProvider = new AlmaCodeProvider(instRegAsJson);
 
-            partners =
-                getBibNrList(bibNrFile).stream()
-                    .map(basebibliotekConnection::getBasebibliotek)
-                    .map(baseBibliotek -> new PartnerConverter(almaCodeProvider, illServer, baseBibliotek).toPartners())
-                    .flatMap(List::stream)
-                    .collect(Collectors.toList());
-            return toIntExact(partners.stream()
-                                  .filter(this::sendToAlma)
-                                  .count());
+            var basebiblioteks = new ArrayList<BaseBibliotek>();
+            var bibnrList = getBibNrList(bibNrFile);
+            for (String bibnr : bibnrList) {
+                try {
+                    basebiblioteks.add(basebibliotekConnection.getBasebibliotek(bibnr));
+                } catch (Exception e) {
+                    logger.info("Could not fetch basebibliotek", e);
+                }
+            }
+            for (BaseBibliotek baseBibliotek : basebiblioteks) {
+                try {
+                    partners.addAll(new PartnerConverter(almaCodeProvider, illServer, baseBibliotek).toPartners());
+                } catch (Exception e) {
+                    logger.info("Could not convert to partner", e);
+                }
+            }
+            var counter = 0;
+            for (Partner partner : partners) {
+                try {
+                    if (sendToAlma(partner)) {
```

Comment:
sendToAlma() kalles to ganger for hver partner?

Reply:
fikset